controllers/pipelines/provider_controller.go

```diff
@@ -0,0 +1,219 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	config "github.com/sky-uk/kfp-operator/apis/config/v1alpha6"
+	. "github.com/sky-uk/kfp-operator/apis/pipelines"
+	pipelinesv1 "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
+	"github.com/sky-uk/kfp-operator/controllers/pipelines/internal/logkeys"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	AppLabel               = "app"
+	ResourceHashAnnotation = "resource-hash"
+)
+
+type ProviderReconciler struct {
+	ResourceReconciler[*pipelinesv1.Provider]
+	Scheme *runtime.Scheme
+}
+
+func NewProviderReconciler(ec K8sExecutionContext, config config.KfpControllerConfigSpec) *ProviderReconciler {
+	return &ProviderReconciler{
+		ResourceReconciler: ResourceReconciler[*pipelinesv1.Provider]{
+			EC:     ec,
+			Config: config,
+		},
+		Scheme: ec.Scheme,
+	}
+}
+
+//+kubebuilder:rbac:groups=pipelines.kubeflow.org,resources=providers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=pipelines.kubeflow.org,resources=providers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=pipelines.kubeflow.org,resources=providers/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
```

Comment:
It seems the blank line here isn't redundant after all (which is why we have it in all other controllers as well). `make manifests` doesn't pick up the provider-related resources it needs to generate if the blank line isn't there. It's a bit silly, but we need it 🤷‍♂️